src/controllers/auth.controller.js

```diff
@@ -0,0 +1,42 @@
+import bcrypt from 'bcrypt';
+import jwt from 'jsonwebtoken';
+import dotenv from 'dotenv';
+import User from '../models/user.js';
+import { RefreshTokenSchema } from '../models/refreshToken.js';
+import handleResponse from './handleResponse.js';
+
+dotenv.config();
+
+const AuthHandler = async (req, res) => {
+	const user = await User.findOneBy({ email: req.body.email });
+	if (!user) {
+		return handleResponse(res, 404, { message: res.__('user does not exist') });
+	}
+	try {
+		if (await bcrypt.compare(req.body.password, user.password)) {
+			const accessToken = jwt.sign(
+				{ user: { id: user.id, role: user.role } },
```

Comment:
@samuel-shyaka-dus on this part, after checking on my end. It would be better if we sign the user object without the parent key.
Since we are assigning those values on the requests as **`req.user=user`** in JWTVerification middleware.

All you have to do it to change `{user: {}}` to `{id: id, role:role}`

Everything else is great here.